File: ios/Bulb.m

//#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface
  RCT_EXTERN_MODULE(Bulb, NSObject)
  RCT_EXTERN_METHOD(toggleOn: (RCTResponseSenderBlock)callback)
  RCT_EXTERN_METHOD(getStatus: (RCTResponseSenderBlock)callback)
@end

File: ios/Bulb.swift
import Foundation

@objc(Bulb) class Bulb: NSObject {
  @objc static var isOn = false
 
  @objc func toggleOn(_ callback: RCTResponseSenderBlock) {
    Bulb.isOn = !Bulb.isOn
    callback([ Bulb.isOn ])
  }
  
  @objc func getStatus(_ callback: RCTResponseSenderBlock) {
    callback([ Bulb.isOn ])
  }
  
  @objc static func requiresMainQueueSetup() -> Bool {  return true }
}
